src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,293 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
-        return ReadActiveModeThreshold(aPath.mEndpointId, aEncoder);
+        return aEncoder.Encode(mICDConfigurationData.GetActiveModeThreshold().count());
 
-    case IcdManagement::Attributes::FeatureMap::Id:
-        return ReadFeatureMap(aPath.mEndpointId, aEncoder);
 #if CHIP_CONFIG_ENABLE_ICD_LIT
     case IcdManagement::Attributes::OperatingMode::Id:
-        return ReadOperatingMode(aPath.mEndpointId, aEncoder);
+        return ReadOperatingMode(aEncoder);
 #endif // CHIP_CONFIG_ENABLE_ICD_LIT
-#if CHIP_CONFIG_ENABLE_ICD_CIP
-    case IcdManagement::Attributes::RegisteredClients::Id:
-        return ReadRegisteredClients(aPath.mEndpointId, aEncoder);
 
-    case IcdManagement::Attributes::ICDCounter::Id:
-        return ReadICDCounter(aPath.mEndpointId, aEncoder);
+    case IcdManagement::Attributes::UserActiveModeTriggerHint::Id:
+        return aEncoder.Encode(mUserActiveModeTriggerBitmap);
 
-    case IcdManagement::Attributes::ClientsSupportedPerFabric::Id:
-        return ReadClientsSupportedPerFabric(aPath.mEndpointId, aEncoder);
+    case IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id:
+        return aEncoder.Encode(
+            CharSpan(mUserActiveModeTriggerInstruction,
+                     strnlen(mUserActiveModeTriggerInstruction, IcdManagement::kUserActiveModeTriggerInstructionMaxLength)));
 
-    case IcdManagement::Attributes::MaximumCheckInBackOff::Id:
-        return ReadMaximumCheckInBackOff(aPath.mEndpointId, aEncoder);
-#endif // CHIP_CONFIG_ENABLE_ICD_CIP
+    default:
+        return Protocols::InteractionModel::Status::UnsupportedAttribute;
     }
-
-    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ICDManagementAttributeAccess::ReadIdleModeDuration(EndpointId endpoint, AttributeValueEncoder & encoder)
+CHIP_ERROR ICDManagementCluster::Attributes(const ConcreteClusterPath & path,
+                                            ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
 {
-    return encoder.Encode(mICDConfigurationData->GetIdleModeDuration().count());
+    AttributeListBuilder attributeListBuilder(builder);
+    const DataModel::AttributeEntry kMandatoryAttributes[] = {
+        IcdManagement::Attributes::IdleModeDuration::kMetadataEntry,
+        IcdManagement::Attributes::ActiveModeDuration::kMetadataEntry,
+        IcdManagement::Attributes::ActiveModeThreshold::kMetadataEntry,
+    };
+    BitFlags<IcdManagement::Feature> featureMap = mICDConfigurationData.GetFeatureMap();
+    const bool hasCIP                           = featureMap.Has(IcdManagement::Feature::kCheckInProtocolSupport);
+    const bool hasUAT                           = featureMap.Has(IcdManagement::Feature::kUserActiveModeTrigger);
+    const bool hasLIT                           = featureMap.Has(IcdManagement::Feature::kLongIdleTimeSupport);
+
+    const AttributeListBuilder::OptionalAttributeEntry optionalEntries[] = {
+        { hasCIP, IcdManagement::Attributes::RegisteredClients::kMetadataEntry },
```

Comment:
So this part is a bit weird.  We have a whole separate class for the CIP-enabled case, no?  Why are we doing this in the base class instead of only doing it in the CIP-enabled case.  If we're not in that case, hasCIP is necessarily false here... though that should be enforced in Startup, perhaps, if we can manage it.  Followup is OK for that enforcement, I guess.

Right now the way this is written up we can claim to have a RegisteredClients attribute and then totally fail to read it.